### DISCUSSION TITLE
Unable to add new dependencies

### DISCUSSION THREAD

**Discussion by bryhasagithub** [pnpm/pnpm#8757] [Q&A] [answered]
So pnpm install works fine but the moment I try to add a new dependency, I will get an error because it tries to treat a few of my packages in the monorepo as external

whether I add it via 
`pnpm add -w @google-cloud/bigquery@6.0.3`
OR 
by just adding it to the package.json as needed, then running `pnpm install`
 
 I'll get the same issue. However, it's not an issue for most of my colleagues. I've tried wiping my repo, pnpm globally, deleted cache files and hidden pnpm related files but still get the same behavior after reinstalling everything.
 
 Would anybody happen to have any ideas on what could be the issue?
 
 
 
![Screenshot 2024-11-14 at 10 23 25 AM](https://github.com/user-attachments/assets/63911e58-4e53-4b3e-9724-31463a7bce28)

**Comment by zkochan** [comment]
It seems like on the screenshot you show a different `package.json` file, not the one that is in the error message. Check the one at `packages/data-editor-ui`.

**Comment by bryhasagithub** [comment]
![Screenshot 2024-11-14 at 12 00 34 PM](https://github.com/user-attachments/assets/bb4ec6a5-a633-40ff-b146-3b863705ec86)
`data-editor-ui` is one of many of our packages that use the `@glide` alias. They are all in the same repo and like I mentioned, have no issue installing when I am not adding any new dependencies but when I do want to add them, it doesn't recognize our packages in the mono repo like `data-editor-ui` as a local package

**Reply by zkochan** [reply, marked as answer]
The version specs are incorrect in some of the dependencies. It should be `workspace:*` with a colon, not `workspace*`.

**Reply by bryhasagithub** [reply]
omg. how it works for everybody else on my team but me is incredible hahahah but this works! tyty

**Reply by zkochan** [reply]
I don't know.

I guess it would be good to extend the error message with information about what version specifier was used for the dependency that wasn't found.
